addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,345 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.rotelra1x.internal.ConfigurationError;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.PortInUseException;
+import gnu.io.RXTXPort;
+import gnu.io.SerialPort;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * The {@link RotelRa1xHandler} handles commands and state changes, and communicates
+ * with the amplifier over the serial port interface..
+ *
+ * @author Marius Bjørnstad - Initial contribution
+ */
+public class RotelRa1xHandler extends BaseThingHandler {
+
+    private static final int BAUD = 115200;
+    private static final long ERROR_RETRY_DELAY_MS = 60000;
+    private int maximumVolume;
+    private RXTXPort serialPort;
+
+    private boolean exit;
+    private volatile boolean power;
+
+    private final Logger logger = LoggerFactory.getLogger(RotelRa1xHandler.class);
+
+    public RotelRa1xHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        maximumVolume = ((BigDecimal) getThing().getConfiguration().get("maximum-volume")).intValue();
+        exit = false;
+        try {
+            connect();
+            Executors.newSingleThreadScheduledExecutor().schedule(() -> {
+                inputLoop();
+            }, 4, TimeUnit.SECONDS);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            disconnect();
+        } catch (ConfigurationError e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            disconnect();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        exit = true;
+        disconnect();
+    }
+
+    private void connect() throws IOException, ConfigurationError {
+        // Note: connect may leave the port open even if it throws an exception.
+        if (serialPort == null) {
+            String portName = (String) getThing().getConfiguration().get("port");
+            if (portName == null) {
+                throw new ConfigurationError("Serial port name not configured");
+            }
+            try {
+                serialPort = new RXTXPort(portName);
+            } catch (PortInUseException e) {
+                throw new IOException(e);
+            }
+            try {
+                serialPort.setSerialPortParams(BAUD, SerialPort.DATABITS_8, SerialPort.STOPBITS_1,
+                        SerialPort.PARITY_NONE);
+            } catch (UnsupportedCommOperationException e) {
+                serialPort.close();
+                throw new IOException(e);
+            }
+
+            // Don't need continuous updates of the display, we still get updates when
+            // the volume, etc., changes
+            serialPort.getOutputStream().write("display_update_manual!".getBytes(StandardCharsets.US_ASCII));
```

Comment:
I guess this write and the one below should call your send method instead of directly writing. This keeps all the writing in one place.

Reply:
The problem is that my current send() functions call connect(). Maybe that's not necessary, and I can instead connect() in handleCommand() before sending. Then I could definitely use send() functions here. That would require multiple changes, but I'll give it a go.

Reply:
On second thought, it seems there is no need to connect inside send() anyway. Commands won't be delivered to the binding unless the thing is ONLINE, so there's no need to trigger a retry of the connection. This makes things easier.